### PR TITLE
Setup test environment

### PR DESCRIPTION
**Pull request by pdany1116**
It is a big PR, sorry. But nothing is broken, trust me. 🏳️‍🌈 We can write tests now!

Look over the test to get familiar with how we should create them.

* Created test helpers with suite setup and teardown
* Create Collector Mock
* Removed utils methods for settings and created propper Settings class where configurations are kept for the entire application lifetime
* Changed configuration files from XML to YAML (easier to parse)
* Removed global objects declared -> used dependency injection (ugly, but does the job for now)

### CODE REVIEW

File: test/settings_test.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from is_iot_sink.settings import Settings
+from threading import Lock
+
+def filepath():
+    return os.getenv('PROJECT_PATH') + '/test/fixtures/settings_test_setup.yml'
+
+def test_get_simple_text_setting():
+    settings = Settings(filepath = filepath(), mutex = Lock())
+
+    assert settings.get('name') == 'SinkTest'
```
Comment (luciiii6) [inline test/settings_test.py:12]: What happens if you call get with an attribute that doesn't exist in the YML file?

File: is_iot_sink/sink.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+        self.__queue_head = Queue(maxsize = 0)
+        self.__running = False
+        self.__mongo_client = MongoClient(self.__settings)
+        self.__valve_manager = ValveManager(self.__settings, self.__mongo_client)
+        self.__allowed_collectors = AllowedCollectors(self.__settings)
+        self.__mqtt_client = MQTTClient(self.__settings)
+        self.__mqtt_client.attach_queue(self.__queue_head)
+        self.__irrigation_factory = IrrigationFactory(self.__settings, self.__valve_manager, self.__mongo_client, self.__allowed_collectors)
+        self.__irrigation = self.__irrigation_factory.create(IrrigationMode.str_to_mode(self.__settings.get("irrigation/initialMode").upper()))
```
Comment (luciiii6) [inline is_iot_sink/sink.py:31]: :+1:

**Review by DariusCr — APPROVED** [review]
looks gucci to me

File: is_iot_sink/sink.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import json
+import os
+
+class Sink:
+    def __init__(self, settings=None):
+        if settings == None:
+            self.__settings_mutex = Lock()
+            self.__settings = Settings(os.getenv('PROJECT_PATH') + '/setup.yml', self.__settings_mutex)
+        else:
+            self.__settings = settings
```
Comment (luciiii6) [inline is_iot_sink/sink.py:21]: Discussed f2f, but I'll write this to see as a reference.

**Pull request merged**